week_3/project/week_3.py

```diff
@@ -13,64 +15,91 @@
     op,
     sensor,
     static_partitioned_config,
+    schedule
 )
 from project.resources import mock_s3_resource, redis_resource, s3_resource
 from project.sensors import get_s3_keys
 from project.types import Aggregation, Stock
 
 
-@op
-def get_s3_data():
-    # Use your ops from week 2
-    pass
-
-
-@op
-def process_data():
-    # Use your ops from week 2
-    pass
-
-
-@op
-def put_redis_data():
-    # Use your ops from week 2
-    pass
+@op(
+    config_schema={"s3_key": str},
+    required_resource_keys={"s3"},
+    out={"stocks": Out(dagster_type=List[Stock])},
+    tags={"kind": "s3"},
+    description="Getting a list of stocks from an S3 file",
+)
+def get_s3_data(context):
+    key_name = context.op_config["s3_key"]
+    output = list()
+    for record in context.resources.s3.get_data(key_name):
+        stock = Stock.from_list(record)
+        output.append(stock)
+    return output
+
+
+@op(
+    ins={"stock_list": In(dagster_type=List[Stock])},
+    out={"agg": Out(dagster_type=Aggregation)},
+    description="Find the highest high value from a list of stocks"
+    )
+def process_data(stock_list) -> Aggregation:
+    high_stock = max(stock_list, key=attrgetter('high'))
+    agg = Aggregation(date=high_stock.date, high=high_stock.high)
+    return agg
+
+
+@op(
+    required_resource_keys={"redis"},
+    ins={"aggregation": In(dagster_type=Aggregation)},
+    description="Put Aggregation data into Redis",
+    tags={"kind": "redis"}
+    )
+def put_redis_data(context, aggregation: Aggregation):
+    date = str(aggregation.date)
+    high = str(aggregation.high)
+    context.log.info(f"Putting stock {date} with high of {high} into Redis")
+    context.resources.redis.put_data(date, high)
 
 
 @graph
 def week_3_pipeline():
-    # Use your graph from week 2
-    pass
+    put_redis_data(process_data(get_s3_data()))
 
 
 local = {
     "ops": {"get_s3_data": {"config": {"s3_key": "prefix/stock_9.csv"}}},
 }
 
-
-docker = {
-    "resources": {
-        "s3": {
-            "config": {
-                "bucket": "dagster",
-                "access_key": "test",
-                "secret_key": "test",
-                "endpoint_url": "http://host.docker.internal:4566",
-            }
+def config_setup(config_key: str):
```

Comment:
Interesting to see you your approach here to allow for dynamic configuration. I unpacked the `docker` var and then overrode the `ops` line.